<a href="https://colab.research.google.com/github/davlsb/data_analytics_2/blob/main/Python/Unit3_Python_Advanced_Pandas_DataWrangling_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Assignment 1 : Python Data Analysis using salaries dataset

# Data Exploration:  This should include summary statistics, means, medians, quartiles, or any other relevant information about the data set. 

In [4]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [6]:
df = pd.read_csv('https://raw.githubusercontent.com/CunyLaguardiaDataAnalytics/datasets/master/Salaries.csv')

In [ ]:
type(df)

pandas.core.frame.DataFrame

In [ ]:
# 397 rows = 397 employees
df.shape

(397, 7)

In [ ]:
df.head()

In [ ]:
df.tail()

In [ ]:
df.describe()

# Mean/median of all salaries

In [18]:
# Mean of all salaries
df["salary"].mean()

113706.45843828715

In [19]:
# Median of all salaries
df["salary"].median()

107300.0

# Median salary between gender and/or rank(title) Etc...


Making a new Data Frame that only has the columns 'sex' and 'salary'

In [20]:
ndf = df.filter(items=["sex","salary"])

In [ ]:
ndf.groupby(["sex",]).median()

salary
sex           
Female  103750
Male    108043

Finding the Median salary for all males

In [22]:
m_df = ndf[df['sex']=='Male'].median().to_string()

In [ ]:
print("the median ", m_df)

the median  salary    108043.0


Finding the Median salary for all females

In [21]:
f_df = ndf[df['sex']=='Female'].median().to_string()

In [ ]:
print("the median ", f_df)

the median  salary    103750.0


Comparing the Mean salary of Males and Females

In [23]:
meanm_df = ndf[df['sex']=='Male'].mean()

In [24]:
meanf_df = ndf[df['sex']=='Female'].mean()

In [25]:
# The difference between mean salaries of male and female
meandif = float(meanf_df-meanm_df)
if(meandif < 0):
  meandif = meandif * -1
print("the difference is", meandif)

the difference is 14088.008738003162


# Data Wrangling – perform some basic transformations on the data

People who have 20 years of experience (Total: 157), (Mean Salary: 122,520.66)

In [28]:
servalary = df[['salary', 'yrs.service']]
servalary

there are  salary         397
yrs.service    397
dtype: int64 workers with over 20 years


In [43]:
over20 = servalary[servalary['yrs.service'] >= 20]
over20count = over20[['salary']]
print("there are ", int(over20count.count()), "workers with over 20 years")
print("their mean salary is ",float(over20count.mean()))

there are  157 workers with over 20 years
their mean salary is  122520.65605095541


In [11]:
over20

salary  yrs.service
3    115000           39
4    141500           41
6    175000           23
7    147765           45
8    119250           20
..      ...          ...
384  125192           21
386  139219           27
388  119450           36
392  103106           30
394  101738           25

[157 rows x 2 columns]

Comparing the salaries of people who have 20+ years of experience vs less than 20 years of experience

In [12]:
under20 = servalary[servalary['yrs.service'] < 20]

In [16]:
saldif = float(under20.mean()[['salary']] - over20.mean()[['salary']])
if (saldif < 0):
  saldif = saldif * -1
print("the difference is ", saldif)

the difference is  14580.151884288745


People who have salary of greater than or equal to 200,000

In [27]:
g200k = df[df['salary'] >= 200000]
print(g200k)

     Unnamed: 0  rank discipline  yrs.since.phd  yrs.service   sex  salary
43           44  Prof          B             38           38  Male  231545
249         250  Prof          A             29            7  Male  204000
364         365  Prof          A             43           43  Male  205500


Conclusions:
There are 3 individuals with a salary above 200k. All 3 of them are males. Moreover, they are all professors. 


# Rename values, subset columns, etc

In [45]:
df.columns = ['renamed1', 'renamed2','renamed3', 'renamed4','renamed5', 'renamed6', 'renamed7']

In [46]:
df

renamed1  renamed2 renamed3  renamed4  renamed5 renamed6  renamed7
0           1      Prof        B        19        18     Male    139750
1           2      Prof        B        20        16     Male    173200
2           3  AsstProf        B         4         3     Male     79750
3           4      Prof        B        45        39     Male    115000
4           5      Prof        B        40        41     Male    141500
..        ...       ...      ...       ...       ...      ...       ...
392       393      Prof        A        33        30     Male    103106
393       394      Prof        A        31        19     Male    150564
394       395      Prof        A        42        25     Male    101738
395       396      Prof        A        25        15     Male     95329
396       397  AsstProf        A         8         4     Male     81035

[397 rows x 7 columns]

# Create some simple visualizations showing some simple statistics

Your observations from the visuals: 

# Draw some conclusions based on the data.  What did you discover?

Based on the analysis performed above, the conclusions are as follows:

* Females make less on average than males, with an average difference of 14088.
* Those with over 20 years of experience make around $14580 more than those with under 20 years.
* Those with a salry above 200k are all professors and male. 


I also discovered that the -

 Your Findings here...



